<launch>

  <node name="controller"
        pkg ="joy"
        type="joy_node"
        respawn="false"
        output="screen">
    <remap from="/joy"
             to="/hyperion/joy"/>
    <param name="dev" value="/dev/input/js1"/>
  </node>

 <node name="hyperion"
       pkg ="hyperion_labyrinth"
       type="hyperion_labyrinth"
       respawn="false"
       output="screen">
    <param name="kp"            type="double"      value="0.21"/>
    <param name="ki"            type="double"      value="0.0"/>
    <param name="kd"            type="double"      value="0.25"/>
    <param name="pid_freq"      type="double"      value="500"/>
    <param name="max_speed"     type="int"         value="100"/>
    <param name="turn_speed_saturation" type="int" value="0"/>
    <param name="stop_button"   type="int"         value="5"/>
    <param name="pid_tunning"   type="int"         value="0"/>
    <param name="tunning_inc"   type="double"      value="0.01"/>
    <param name="pid_mean_size" type="int"         value="4"/>
    <param name="error_calculation" type="int"     value="2"/>
    <param name="center_dist"   type="double"      value="170.0"/>
    <param name="front_wall_dist"   type="double"  value="80.0"/>
  </node>

  <node name="ultrasonic_sensors"
        pkg ="ultrasonido"
        type="ultrasonidos.py"
        respawn="false"
        output="screen">
    <remap from="/stop_topic"
           to="/hyperion/stop"/>
    <remap from="/ultrasonido_topic"
           to="/hyperion/distances"/>
  </node>

  <node name="motor_driver"
        pkg ="controlador_motores"
        type="motor_driver.py"
        respawn="false"
        output="screen">
    <remap from="/motor_speeds"
           to="/hyperion/speeds"/>
    <remap from="/turn"
           to="/hyperion/turn"/>
  </node>


</launch>
